src/interfaces/coral_web/src/components/Agents/CreateAgentForm.tsx

```diff
@@ -0,0 +1,141 @@
+import { useRouter } from 'next/router';
+import React, { useContext, useState } from 'react';
+
+import { AgentForm, AgentFormFieldKeys, AgentFormFields } from '@/components/Agents/AgentForm';
+import { Button, Text } from '@/components/Shared';
+import { DEPLOYMENT_COHERE_PLATFORM, MODEL_COMMMAND_R_PLUS } from '@/constants';
+import { ModalContext } from '@/context/ModalContext';
+import { useCreateAgent, useIsAgentNameUnique } from '@/hooks/agents';
+import { useNotify } from '@/hooks/toast';
+import { useParamsStore } from '@/stores';
+
+/**
+ * @description Form to create a new agent.
+ */
+export const CreateAgentForm: React.FC = () => {
+  const router = useRouter();
+  const { open, close } = useContext(ModalContext);
+  const { error } = useNotify();
+  const { mutateAsync: createAgent } = useCreateAgent();
+  const {
+    params: { preamble },
+  } = useParamsStore();
+  const isAgentNameUnique = useIsAgentNameUnique();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [fields, setFields] = useState<AgentFormFields>({
+    name: '',
+    description: '',
+    preamble,
+    deployment: DEPLOYMENT_COHERE_PLATFORM,
+    model: MODEL_COMMMAND_R_PLUS,
+    tools: [],
+  });
+
+  const fieldErrors = {
+    ...(isAgentNameUnique(fields.name) ? {} : { name: 'Assistant name must be unique' }),
+  };
+
+  const canSubmit = (() => {
+    const { name, deployment, model } = fields;
+    const requredFields = { name, deployment, model };
+    return Object.values(requredFields).every(Boolean) && !Object.keys(fieldErrors).length;
+  })();
+
+  const handleChange = (key: Omit<AgentFormFieldKeys, 'tools'>, value: string) => {
+    setFields({
+      ...fields,
+      [key as string]: value,
+    });
+  };
+
+  const handleToolToggle = (toolName: string, checked: boolean) => {
+    const enabledTools = [...(fields.tools ? fields.tools : [])];
+    setFields({
+      ...fields,
+      tools: checked ? [...enabledTools, toolName] : enabledTools.filter((t) => t !== toolName),
+    });
+  };
+
+  const handleOpenSubmitModal = () => {
+    open({
+      title: `Create ${fields.name}?`,
+      content: (
+        <SubmitModalContent
+          agentName={fields.name}
+          onSubmit={handleSubmit}
+          isSubmitting={isSubmitting}
+          onClose={close}
+        />
+      ),
+    });
+  };
+  const handleSubmit = async () => {
+    if (!canSubmit) return;
+
+    try {
+      setIsSubmitting(true);
+      const agent = await createAgent(fields);
+      setFields({
+        name: '',
+        description: '',
+        preamble,
+        deployment: '',
+        model: '',
+        tools: [],
+      });
+      close();
+      setIsSubmitting(false);
+      router.push(`/agents?id=${agent.id}`, undefined, { shallow: true });
```

Comment:
```suggestion
      router.push(`/agents?assistantId=${agent.id}`, undefined, { shallow: true });
```

Reply:
this was recently changed because `router.query.id` would conflict with query params and pathname params for some reason =/

This will conflict for example
`/c/[:id]?id=test-id`